server.js

```diff
@@ -338,9 +339,22 @@ function isValidateOn(document) {
 	return validateLanguages.includes(document.languageId);
 }
 
-connection.onInitialize(() => {
+/** Whether or not dynamic registration for document formatting should be attempted. */
+let registerFormatterDynamically = false;
+
+/**
+ * A promise that resolves to the disposable for the dynamically registered document formatter.
+ * @type {Promise<import('vscode').Disposable> | undefined}
```

Comment:
Is this type not included in some non-vscode specific library?

I know this is just a type annotation so it won't break runtime but it's still not exactly correct to use it on the server side.

Reply:
After a closer look, [`vscode`'s `Disposable` type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8c66ac2/types/vscode/index.d.ts#L1483-L1507) is actually different from the one that `vscode-languageserver` exports, which it re-exports from [`vscode-jsonrpc`](https://github.com/microsoft/vscode-languageserver-node/blob/release/jsonrpc/5.1.0-next.1/jsonrpc/src/events.ts#L7-L12). So this line really should be using the `vscode-languageserver` type. Looks like there are no type errors due to the types' overlap.